components/bl-audio-recorder-component/src/index.js

```diff
@@ -32,8 +35,14 @@ export default function AudioRecorder({ component, eventHandlers, elRef }) {
     if (state) {
       onStateChange({ state });
     }
+
+    simpleTimer(state, StreamState.RECORDING, setTime, setTimerInterval, timerInterval);
```

Comment:
It could be an isolated class Timer that accepts a time setter(setTime), and provide methods play/pause or toggleState(<state>). In this case, the component will accept a set event of time changing, and have controls to play and pause the timer, but doesn't know about any intervals and logic that lie under it.